Rest.Services/Repositories/ProductRepository.cs

```diff
@@ -0,0 +1,49 @@
+using Rest.Models;
+namespace Rest.Services;
+public class ProductRepository : IRepository<long, Product>
+{
+    private Dictionary<long, Product> _productDictionary;
+    private long _id;
+
+    public ProductRepository()
+    {
+        _productDictionary = new Dictionary<long, Product>();
+        _id = 0;
+    }
+
+    public void Add(Product entity)
+    {
+        _id++;
+        Product newProduct = entity with
+        {
+            Id = _id,
+            Name = entity.Name,
+            Difficulty = entity.Difficulty
```

Comment:
Why do we have a difficulty here?

Reply:
Because I initially declared the product entity to have these three properties. So, to initialize a product I'm required to pass these. Or do you mean something different?

Reply:
A product in my project, is a finished meal or a dish. So, I added a difficulty property to take it into account when a chef had to prepare it. It isn't used now but this will be used in the near future.

Reply:
I see cool thanks